src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/HenleggBehandlingService.kt

```diff
@@ -0,0 +1,98 @@
+package no.nav.familie.ks.sak.kjerne.behandling
+
+import no.nav.familie.ks.sak.api.dto.HenleggBehandlingDto
+import no.nav.familie.ks.sak.api.dto.HenleggÅrsak
+import no.nav.familie.ks.sak.api.dto.ManueltBrevDto
+import no.nav.familie.ks.sak.common.exception.Feil
+import no.nav.familie.ks.sak.common.exception.FunksjonellFeil
+import no.nav.familie.ks.sak.config.FeatureToggleConfig
+import no.nav.familie.ks.sak.config.FeatureToggleService
+import no.nav.familie.ks.sak.integrasjon.oppgave.OppgaveService
+import no.nav.familie.ks.sak.kjerne.behandling.domene.Behandling
+import no.nav.familie.ks.sak.kjerne.behandling.domene.BehandlingRepository
+import no.nav.familie.ks.sak.kjerne.behandling.domene.BehandlingStatus
+import no.nav.familie.ks.sak.kjerne.behandling.steg.StegService
+import no.nav.familie.ks.sak.kjerne.brev.BrevService
+import no.nav.familie.ks.sak.kjerne.brev.domene.maler.Brevmal
+import no.nav.familie.ks.sak.kjerne.logg.LoggService
+import no.nav.familie.ks.sak.sikkerhet.SikkerhetContext
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+class HenleggBehandlingService(
+    private val stegService: StegService,
+    private val featureToggleService: FeatureToggleService,
+    private val brevService: BrevService,
+    private val oppgaveService: OppgaveService,
+    private val loggService: LoggService,
+    private val behandlingRepository: BehandlingRepository
+) {
+
+    @Transactional
+    fun henleggBehandling(behandlingId: Long, henleggBehandlingDto: HenleggBehandlingDto) {
```

Comment:
Jeg synes at jo mindre forhold en service har til en dto'er, jo bedre.
Går det ann å unpacke dto'en her slik at henleggBehandling tar inn en årsak og beskrivelse istedenfor dto?
Da får vi kutta ut avhengigheten til request objekt.

```suggestion
    fun henleggBehandling(behandlingId: Long, årsak: HenleggÅrsak, begrunnelse: String) {
```

På den måten så får man også kalt denne metoden fra andre servicer senere uten å måtte opprette dto objekt.